File: server/php/src/Application/Domain/Payments/PaymentDetails/ReattemptInstructions.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class ReattemptInstructions
{
    public ?ReattemptInstructionsConditions $conditions = null;
    public ?int $frozenPeriod = null;
    public ?string $indicator = null;

    public function toArray(): array
    {
        return [
            'conditions' => ($this->conditions ?? new ReattemptInstructionsConditions())->toArray(),
            'frozenPeriod' => $this->frozenPeriod,
            'indicator' => $this->indicator,
        ];
    }
}

File: server/python/app/application/domain/payments/payment_details/reattempt_instructions.py
from dataclasses import dataclass
from typing import Optional

from app.application.domain.payments.payment_details.reattempt_instructions_conditions import ReattemptInstructionsConditions

@dataclass
class ReattemptInstructions:
    conditions: Optional[ReattemptInstructionsConditions] = None
    frozen_period: Optional[int] = None
    indicator: Optional[str] = None